app/handlers/sessions_create.rb

```diff
@@ -47,12 +47,12 @@ def authorized?
 
   def handle
     outputs[:status] =
-      if signing_up?
-        handle_during_signup
-      elsif signed_in?
+      if signed_in?
         handle_while_logged_in
       elsif logging_in?
         handle_during_login
+      elsif signing_up?
+        handle_during_signup
```

Comment:
By changing the order of the checks we are able to avoid the case where a login attempt goes down the signup path.